app/routes/dashboard/purchase/confirm.tsx

```diff
@@ -0,0 +1,86 @@
+import { Outlet, redirect } from 'remix'
+import type { ActionFunction, LoaderFunction } from 'remix'
+import type { Subscription, User } from '@prisma/client'
+import { validatePhoneNumber, validateRequired } from '~/utils/validators'
+import { auth } from '~/services/auth.server'
+import { db } from '~/utils/db.server'
+
+interface TransactionFields {
+  userId: string
+  subscriptionId: string
+  bankName: string
+  bankAccountNumber: string
+  bankAccountName: string
+  amount: number
+  status: string
+}
+
+type ActionData = {
+  formError?: string
+  fieldErrors?: {
+    name: string | undefined
+    phoneNumber: string | undefined
+    bankName: string | undefined
+    bankAccountNumber: string | undefined
+    bankAccountName: string | undefined
+    amount: number | undefined
+    paymentTime: string | undefined
+  }
+  fields: TransactionFields
+}
+
+export const action: ActionFunction = async ({ request }) => {
+  const user = await auth.isAuthenticated(request, {
+    failureRedirect: '/login',
+  })
+
+  const form = await request.formData()
+  const name = form.get('name')
+  const phoneNumber = form.get('phoneNumber')
+  const bankName = form.get('bankName')
+  const bankAccountNumber = form.get('bankAccountNumber')
+  const bankAccountName = form.get('bankAccountName')
+  const amount = form.get('amount')
+  const paymentTime = form.get('paymentTime')
+
+  if (
+    typeof name !== 'string' ||
+    typeof phoneNumber !== 'string' ||
+    typeof bankName !== 'string' ||
+    typeof bankAccountNumber !== 'string' ||
+    typeof bankAccountName !== 'string' ||
+    typeof amount !== 'number' ||
+    typeof paymentTime !== 'string'
+  ) {
+    return { formError: 'Form not submitted correctly.' }
+  }
+
+  const fieldErrors = {
+    name: validateRequired('Nama Lengkap', name),
+    phoneNumber: validatePhoneNumber('Nomor WhatsApp', phoneNumber),
+    bankName: validateRequired('Nomor WhatsApp', bankName),
+    bankAccountNumber: validateRequired('Nama Bank', bankAccountNumber),
+    bankAccountName: validateRequired('Nomor Rekening', bankAccountName),
+    amount: validateRequired('Nominal', amount),
+    paymentTime: validateRequired('Nominal', paymentTime),
+  }
+
+  const fields: TransactionFields = {
+    userId: user.id,
+    subscriptionId: '123', //should create assign subcription first?
```

Comment:
In PR #91, I updated the Transaction table not related to the Subscription Entity. Hence you don't need to create the Subscription record first in this case.

Therefore, I think you need to do #98 first before proceeding with this ticket. Would you be able to do it? If not, I can try to do it by myself later today. 😉

Reply:
Of course i'd like to, i will create the draft pr soon. Thank you for the help, Mas @zainfathoni !

Reply:
Since we unblocked #98 just now, please resume your work for this PR, Mas @AchmadWahyu. Thanks! 🙏